### PR TITLE
Migrate Cypress E2E tests to GitHub Actions

### PR DESCRIPTION
**Pull request by ddzz**
## Description
This PR migrates Cypress E2E tests to GitHub Actions. https://github.com/department-of-veterans-affairs/va.gov-team/issues/25308

## Acceptance criteria
- [ ] Cypress tests pass on GitHub Actions.

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -68,3 +68,32 @@ jobs:
...
+        with:
+          name: cypress-video-artifacts
+          path: cypress/videos
+
+      - name: Archive Cypress test screenshots
+        uses: actions/upload-artifact@v2
+        if: ${{ failure() }}
+        with:
+          name: cypress-screenshot-artifacts
+          path: cypress/screenshots
```
Comment (oseasmoran73) [inline .github/workflows/e2e.yml:99]: nit; EOL

File: config/cypress-reporters.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "reporterEnabled": "spec, mocha-junit-reporter",
+    "mochaJunitReporterReporterOptions": {
+      "mochaFile": "test-results/e2e-test-output-[hash].xml"
+    }
+  }
```
Comment (oseasmoran73) [inline config/cypress-reporters.json:6]: nit; EOL

**Review by U-DON — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -68,3 +68,32 @@ jobs:
         timeout-minutes: 20
         env:
           BUILDTYPE: ${{ matrix.buildtype }}
+
+      - name: Start content-build
+        run: node src/platform/testing/e2e/test-server.js --buildtype vagovprod --port=3002 &
```
Comment (oseasmoran73) [inline .github/workflows/e2e.yml:73]: I notice that you have a matrix strategy and use `${{ matrix.buildtype}} in a some places. Given that the build config is set to `vagovprod` does it make sense to keep matrix? This is only running in one single instance, correct?

Comment (ddzz) [reply .github/workflows/e2e.yml:73]: Yeah, only one instance. I'll take it out, thanks for reviewing!

**Review by timwright12 — APPROVED** [review]

**Review by oseasmoran73 — APPROVED** [review]

**Review by 1Copenut — APPROVED** [review]
LGTM!

**Pull request merged**